### PR TITLE
Move `std-context` to other contexts

### PR DESCRIPTION
**Pull request by yevhenii-nadtochii**
This PR moves`std-context` to `contexts` directory. 

Additionally, `StdContextDataProvider` is tested now with `AbstractContextDataProviderSpec`. This suite is meant to be run upon every implementation of `ContextDataProvider`. I've updated the implementation to make tests pass. 

The following have been done:

1. `StdScopedLoggingContext` is a Kotlin object now, so it is clear that it is a singleton. Previously, it bound to a firstly created instance of `StdContextDataProvider`, which is avoided now. So that every instance of the provider is operational now.
2. `CurrentStdContext` object has been extracted from `StdContextData` to make clear that there is a singleton that stores the current context data and exposes this data to the provider.

### CODE REVIEW

**Comment by yevhenii-nadtochii** [comment]
@armiol @alexander-yevsyukov PTAL

File: contexts/logging-tls-context/README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+## Thread local based logging context for JVM
+
+This module provides the basic implementation of the logging context that
+uses thread local storage (TLS) to store the current context data.
+
+To use it, add `spine-logging-thread-local-context` to `runtimeOnly` configuration:
+
+```kotlin
+dependencies {
+    runtimeOnly("io.spine:spine-logging-thread-local-context:$version")
```
Comment (armiol) [inline contexts/logging-tls-context/README.md:10]: Do we say "tls" or "thread local"?

Please recheck. And if there are two different names for the same thing (say, module name and artifact name), let's document it.

**Review by armiol — CHANGES_REQUESTED** [review]
@yevhenii-nadtochii in general, I think there is not a big deal to reflect the nature of the "data storage" in the name of an artifact. Basically, logging users just want this whole thing to work properly. And those who really want to dive deep into the details, will go and read the documentation. 

To me, it's the same thing as to call our `InMemoryStorage` with its specifics, e.g. `ShmStorage`, where
"Shm" stands for "synchronized hash map".

**Comment by yevhenii-nadtochii** [comment]
@armiol Yes, maybe it is better to roll back to `std`. 

To me, it is either `std-context` or `tls-context`, but I haven't come up with other options. Its implementation has nothing to reflect in the name. Looks like a backend that prints directly to `System.out`, which we would call `std-backend`.

What do you think?

**Comment by armiol** [comment]
@yevhenii-nadtochii to me "tls" is definitely NOT a thread-local storage :) It's for sure transport layer security.

I wouldn't guess the meaning without drilling down into the implementation, ever. Therefore, I would avoid "tls" as much as possible.

**Comment by yevhenii-nadtochii** [comment]
@armiol PTAL

**Review by armiol — APPROVED** [review]

**Pull request merged**